music_assistant/server/controllers/player_queues.py

```diff
@@ -1525,3 +1573,11 @@ def _parse_player_current_item_id(self, queue_id: str, player: Player) -> str |
             if self.get_item(queue_id, current_item_id):
                 return current_item_id
         return None
+
+    async def _schedule_dont_stop_the_music(self, queue: PlayerQueue):
+        """Auto turn on Radio Mode based on enqueued Media Items."""
+        queue.radio_source = [
+            await self.mass.music.get_item_by_uri(uri) for uri in queue.enqueued_media_items
+        ]
+        # schedule the next item in the queue
+        self.mass.create_task(self._fill_radio_tracks(queue.queue_id))
```

Comment:
this is already executed as background task, why spawn another task ?

Reply:
Fair point